FileManipulator/GlobalJsonManipulator.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Messerli.FileManipulatorAbstractions;
+using Messerli.FileSystem;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Messerli.FileManipulator
+{
+    public sealed class GlobalJsonManipulator : IGlobalJsonManipulator
+    {
+        private const string MsBuildSdksJsonProperty = "msbuild-sdks";
+
+        private readonly IFileOpeningBuilder _fileOpeningBuilder;
+
+        public GlobalJsonManipulator(IFileOpeningBuilder fileOpeningBuilder)
+        {
+            _fileOpeningBuilder = fileOpeningBuilder;
+        }
+
+        public async Task ModifyGlobalJson(string filePath, GlobalJsonModification modification)
+            => await WrapExceptions(filePath, async () =>
+            {
+                var document = await GetJsonDocument(filePath);
+                AddMsBuildSdks(document, modification.MsBuildSdksToAdd);
+                await WriteJsonFile(filePath, document);
+            });
+
+        private static async Task WrapExceptions(string filePath, Func<Task> action)
+        {
+            try
+            {
+                await action();
+            }
+            catch (Exception exception)
+            {
+                throw new GlobalJsonManipulationException(exception, filePath);
+            }
+        }
+
+        private static void AddMsBuildSdks(JObject document, IEnumerable<MsBuildSdk> sdks)
+        {
+            var sdkList = document.GetOrInsert(MsBuildSdksJsonProperty, () => new JObject());
+            var sdkListObject = sdkList as JObject
+                ?? throw new MalformedGlobalJsonException(
+                    $"Key '{MsBuildSdksJsonProperty}' is expected to be an object, but is of type {sdkList.Type}");
```

Comment:
Clarify that we mean a JSON object

Reply:
https://github.com/messerli-informatik-ag/meta-generator/pull/14/commits/6869f35d33afa5f80ce9898d03bc12a084638517